ibis/backends/sql/compilers/datafusion.py

```diff
@@ -605,5 +602,22 @@ def visit_Greatest(self, op, *, arg):
     def visit_Least(self, op, *, arg):
         return self.f.least(*arg)
 
+    def visit_BitwiseNot(self, op, *, arg):
+        # https://stackoverflow.com/q/69648488/4001592
+        return sge.BitwiseXor(this=arg, expression=sg.exp.convert(-1))
+
+    def visit_Clip(self, op, *, arg, lower, upper):
```

Comment:
Perhaps an assertion here to verify the invariant that at least one of `lower` or `upper` must be non-`None`?

Alternatively, if both are `None` it could be reasonable to do something like:

```python
if not ifs:
    return arg
return sge.Case(ifs=ifs, default=arg)
```

Reply:
I'll fix this up! The base sql compiler just returns the input if both `lower` and `upper` are `None`.

Reply:
Actually, in that case we should be able to simply remove this override and use the base implementation since datafusion now supports greatest and least.